huaweicloud/services/cdm/resource_huaweicloud_cdm_link.go

```diff
@@ -0,0 +1,310 @@
+package cdm
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/chnsz/golangsdk/openstack/cdm/v1/link"
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/fmtp"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/logp"
+)
+
+const (
+	configPref = "linkConfig."
+)
+
+func ResourceCdmLink() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceCdmLinkCreate,
+		ReadContext:   resourceCdmLinkRead,
+		UpdateContext: resourceCdmLinkUpdate,
+		DeleteContext: resourceCdmLinkDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"cluster_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"connector": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"config": {
+				Type:     schema.TypeMap,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Required: true,
+			},
+
+			"password": {
+				Type:      schema.TypeString,
+				Optional:  true,
+				Sensitive: true,
+			},
+
+			"access_key": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				RequiredWith: []string{"security_key"},
+			},
+
+			"security_key": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Sensitive:    true,
+				RequiredWith: []string{"access_key"},
+			},
+
+			"enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+		},
+	}
+}
+
+func resourceCdmLinkCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	region := config.GetRegion(d)
+	client, err := config.CdmV11Client(region)
+	if err != nil {
+		return diag.Errorf("error creating CDM v1 client, err=%s", err)
+	}
+
+	opts := link.LinkCreateOpts{
+		Links: []link.Link{
+			{
+				Name:             d.Get("name").(string),
+				ConnectorName:    d.Get("connector").(string),
+				Enabled:          utils.Bool(d.Get("enabled").(bool)),
+				LinkConfigValues: buildLinkConfigParamter(d),
+			},
+		},
+	}
+
+	clusterId := d.Get("cluster_id").(string)
+	logp.Printf("[DEBUG] Creating CDM link opts: %#v", opts)
```

Comment:
here, the sensitive fields such as password, ak, sk, etc. will be printed in the log, which are not compliant and not allowed.